### ISSUE TITLE
Can't resize buffers if they were not initialized and were subscribed to earlier by plot~

### ISSUE THREAD

**Issue by balintlaczko** [flucoma/flucoma-max#159]
flucoma-max build from pr/140 after the [MSVC commit](https://github.com/flucoma/flucoma-max/pull/140/commits/9947a70466524a3a3ae2e67beb92bb6c6dcb1ae8).

Example patch:

```
<pre><code>
----------begin_max5_patcher----------
2133.3oc6ajriiai8raf9ePPnOVsAIknVlSYxk9Ttj.L.AMFXPaQ6hokEEjn
pp5IH829vEsaK2xNR1YxnpPY6hOR8V3aiuG8u+92sxdK+MZts0+v5yVqV86x
QVoGSMxppAVYej71tXRtdh1IzW4a+M6mJgInuIziusX+dZ12r3I4TwthrWn0
yIkH18LK4vlL5NgAaXWvZvSVNH0qPPf5Mj7er92UKJo3HKIlJzXE1ZTdgnZX
T0vlwDeMkZd716i4Dg8SRxhjbvt4oxhzDqjC9nSfsdv+38uS8t7sm9SKDRsX
ILwEYbrlwQXeM+69cXbv0x3Jd977KJrKcQyJ4vJVbk8dVL8EZVNimzVpuxlj
l1Z7UsWjRx7ab8yJ3olwXIlwPMikQegU8HZMLISJiDRATQllIreyy0t0ShGQ
yRJXZBpbT0dUEoo2TRHGo4ojclGfZyqFdKQenQYyS8lmQAzM.zR1K22ODy28
EZTaYuTJmRSXIoYzbZhfHJ4gF3Qz8jhXwl87DQN6+noBnR09bSXeIYddnJNQ
yD+yLFItgMNjwh3IJBo6NiZ7JT9YKHVyh3N7jdJIjzysbohgT9LDzbIyVjuk
jo131FS6ryYK373tvZVYLcunDdJKIou.UvSu.zL1gmuzp2xkPOdwGuFT9lhD
C3MRkDwlbxK8j7BRbbosbOL7FIgcjHnBlY+.AZfRSHRF947cY733t7sAzKmC
TjT8eG8UVj3YMx5ndHW.KsRwxtYaOhcflK5MnfbHu2P4huZ1BZOVw1RC8MB5
wzXI2zaFRKGVtH+Y9q4kyrR8qinnIDQGC+1dI6B3hdK65wT5mNR6itMzy32D
EXTsAZyV7oALFNnwY7e1AzI9P6Fy3D+nt10.JicT4S5lkPRuW4jCzyKhxnx.
qVQYEGexR5pfkPyjDH05iPK3XEakuMJ4F5VkaWPl4bmkYF4TJmkHrR3IzJIm
3Y1tujHWojUJGZGOlmYAz+VOXr7OqNYwLX.csBIrLelPmGj.FL8BXC5GG+i7
L4y.pd81rKACwd3NjwN9wixHwc8loWRRD8s1YJbukFk1ays3v6JEGnADGsxR
PowOjedMiplv.BpbdQ1tJkzRejOY0iokA4Drj5rm9bsmgSlIOKxjbJX76hyI
IBt.IBmSRzYzjnmYlyIw.FMwfuHwTAnIIdaUhYQaL4htgHDYrsEBi1Xmymbk
4IIykeKItWlMmMOpZJ5lOQXZLW7sKdvuPiy.e8oPbgZGDd3q8HuvgO424OkK
nlnT4CJox1F4MGhSGEz.QSmMu.638xtNBplZZRhUxzpnsaZxDUE2s0A4TJas
f1K3phM2JyTcvkKOapvXx0GRD+HgkHWIISzKy5ZfzjnSn250UgQEERxN8Iej
j8E4oeMRm1xhR3wjsz3y.VmlVCc8wNX2.7bzU0xFjrLSXPpx.dHhhKOckjle
6DIUIfuVBn4n1kFrSZcRxnxCaF0Nc1KY63KyaEWWzDHL71JWzUa6.u8JD0K6
zFV+CXqO3OFl03gv46wqnIhWQSe0v1GWvhVusXOJVdLyQvyPiyQezTyyZ7ed
91YF4aU8Sx+l0OXBlNxRiVIKbM0FMz61DF2RIBcbmdgg5.9JiAKywXGY4gCa
WkTX4YLleCdr87m.ftvf2wD.7VR.3uJI.Dr100AG5E3DFBb8CvCjN.1YMB4h
c7BBw9XXH14+qxNn1+YiyhFmnp5fY8C6oDU+BtN2IgHiAnys4MwYPKOSALa0
0oALE8eDcbxDS8QzvI+kFNszvokFNszvokFNszvokFNszvokFNszvokFNszv
okFNcGZ3TYYxu2MbBAWp2zRCm9e7FNgB7eDMbBF7HZ3TEydea3DD+Ha3TIO+
.Z3DDb+Z3D4X52s1nURhp1MAtesaJb5kDUeKDFCm6DDr1o8OP2luQBnGv2Hg
fYu6aiQrT4sqz2mq68w22b15spTgt2sdaISn+xjID.r1G365f8.9gP.D6MTl
Q.7ZOnGF436JOUM1O.92xLkjG3QHOUyHrZLtBcqd86YyjRxjm7QpvrwTa5Kz
UI3k6n1YMpbmufF5VKpdIlySWSX6unzIDs1q8OX+5nGtdOfnGy4E3oJtw.sg
M+ZBrT1EVf+CpKrmbotLypa8nJka8KZQkz6jhUD1oVECTmBHnUcJN69znwGd
L3yYxPm6nPme2Y0qpZ2AJ.N.A.m.BnO2MfL.+3IA2YbavYbJddSllmy3T8lN
UcGmQgPzzgPznPHb5PH7NuGhBuFDNKlNijD.ynoilDfikDlBD5eEAIlGo9nn
.DbNk59iQpilNqKXv8MzLbTdjqnpo.gfQgP7zgvwfuvYTGZTDvXC7WlsYuK3
lF28uXa8uTam4BscgKy1IWjsx9eYt5N8yzs4ZfUDw3+h9dXs4mnIEkYzWcmw
5Iq1dXOKNttBFcZdVUp4kE3n4Bz0tdGnPWHLPUuCGniOxS+I4Gv31lskKBVs
JWraH.olqqGx2Eq+T.xA6nqbR+0gZvF.FZvAHL.3Z9jbHXuhsXKOKQ40Oyu8
EmJMimxypuPdqcBaVQgfeHiDwJ6LN3jd78TkdWlbJmbAprOJEur5U1UWsd6w
bTseLtf9Q3k2ZH61IQSGgsCz2y0Wwzd0BvP0PUr+LPoY7WStARsg.wRkA2P8
N07RpeJiRuERUWsNrtBxpOA8Bj5jyFU9yznafF8Cb0k2VRdtXopu1LQZufBm
MB8Woww7WuAZM.ggZkT+.X.TSo3PrxHusIpc6695m029UbH.DLwrhtznehGe
SBco5JP6sAGhfXMOAa6sYJozTRBM9ew3M2Yn4xksSsAojU.9AFmuRcdsaXzM
4zF92Fm1kw2G3Vh+92Imx+E.m5KV5
-----------end_max5_patcher-----------
</code></pre>
```

### Summary

If the plot~ objects subscribe to the destination buffers of fluid.onsetcurve~ and fluid.ampcurve~ __before__ the process, the fluid objects won't be able to resize the buffers anymore. Solution A: send the refer message to the plot~-s __after__ the process has been done OR initialize the buffers with `@samps 1`, in which case everyone is happy.

### Steps to reproduce

Upon load the plot~-s complain:

![image](https://user-images.githubusercontent.com/45127463/159160964-0d1d3167-5b3b-47b6-9938-1a29492c63fe.png)

After hitting the button we get an unexpectedly long hang (5-8s), then finally:

![image](https://user-images.githubusercontent.com/45127463/159161003-f77a97f9-95a7-49f3-9370-75f1e5cefbfd.png)

### Solution 1: don't subscribe before done:

Patch:
```
<pre><code>
----------begin_max5_patcher----------
2120.3oc6arriiai7rGf4ePPXN1iAIkndrmxlKyobIAXABFrvf1h1MyHSJHQ
0cOaPlu8khTuskGYGI6rYU2nscyhT0CVuXUz+96e2J6sh2nY1V+CqOasZ0uq
FYkdrhQVUMvJ6ij21ESxzSzlSeUr82repDlj9lTO91786ooeyRvynxc4ouPq
mSBQt6YF+vlT5NoAat9AqAOY43V7JDn+GDZMv5eWsHd9QFOlJ0XE1ZTQtrZX
T0vlwjeMgZd716iEDo8SJxhvOX27TYQZhUwAezIvVO3e792U7t5sm9SKDRrX
bl7hLdfgWwgM7+EXbv0x3E7744WTXW5hlVxgUr3J68rX5KzzLlf2Vpuxljjz
Z7UsWTgj42D5mUvSMiw3lwPMikRegU8HZMLIUIijJATdplIreyy0t0SRDQS4
4LMAUNZwdUEoo2T3jizrDxNyCnXyqFdin2Hxg.uh27bAEu4F.ZI6U66GhE69
BMpsrWIkSnbFOIklQ4RhrjGZfGQ2Sxika1K3xL1+QSEP0l5Ymv9Rx77PK3DM
S7OSYj3F13PJKRvKHjt6LEiWgxOaAwZVD2gmzSgSRN2xUJFJ4yPPyTLad1VR
ZwF21XZmcNaoPD2EVyJio6kkvSXbdeApTjbAnorCOeoUuUnfd7hOdMnrM4bC
3MJkD4lLxK8j7RRbbosbOL7FgyNRjTIyref.MPobhhgeNaWpHNtKea.8x4.E
oT+2QekEIeVirNpGpEvRpTrra11iXGnYxdCJIGx5MTl7qlsf1ikuszPeijdL
IVwM8lgxxgkIydV7ZV4LqT+5HJZBQzwvusWxt.tn2xtdLU9oiz9naC8L9MQA
FUaf1rEeZ.igCZbF+mc.chOztwLNwOpqcMfxXGU9jtYIjx6UF4.87hnTpJvp
UTZ9wmrTtJXbZph.oVeDZAGqXq7sQI2P2pb6BxLm6rLyHmRDLtzhK3zJIm7Y
1tuvUqTwJkCsSDKRs.5eqGLV8mUmrXFRBi0JjPS3DmPmGj.FL8BXC5GG+i77
07On50aytDLD6g6PF6DGOphD20aldI7H5asyT3dKMJs2lawg2UJNPCHNZkkP
gF+P940LZwDFPPkIxS2UojV5i7IqdLsJHmjwqyd5y0dFNYlhzHSxofwuKNmj
H3BjHbNIQmQShdlYNmDCXzDC9hDSEflj3sKRLKZiIWzMDoLksMWZzF6b9jqL
OIUt7aIw8xr4r4QUSQ27IBShExucoC9gQlS7F.0m6.pcP3gu1i7BG9jem+Tt
fZhpHePEU11Hu4Pb5nfFHZmVMu.638xtNBplZZRhUwzEQa2zjIZQb2VGjqPY
qEzdAWKXyspLUGb4pylJMlb8gDINRXb0JIoxdYVWCjxiNgdqWWEFKnPR5oO4
ijzunN8qQ5zVVTBOlrkFeFv5zzZnqO1A6FfmitpV1fjkYBCRUFvCQTB0oqTz
7amHoJA70R.MG0tzfcRqSRJUcXyn1oydIaGuPu035hl.gg2V4htZaG3sWgnd
Ym1v5e.a8A+wvrFtz46wqnIhWQSe0v1GmyhVuMeOJVcLyQvyPiyQezTyyZ7e
d91YF46h5mj8MqevDLcjkFsRV3ZpMZn2sILtkRD53NCJ7liuNNduLHYogdH9
No6ism+39P26abeuk39+UItevZWWGbnWfSXHv0O.OPV.Xm0HjK1wKHD6iggX
m+uJofZ2lM9JZ7cVT9KqeXOkTzlfqyaRnqw.z41xZvYPKOScKa0roALE8eDM
Zx3n4QzmI+k9Lszmok9Lszmok9Lszmok9Lszmok9Lszmok9Lszmok9LcO5yj
+inMSH3R4lVZyz+i2lIDB+HZyDL3QzloJl891lIH9Q1loRd9AzlIH390lIxw
juaoQqjDUMYBb+ZxT3zKIp9tGLFN2A4s1o8OP2luGBnGv2CgfYqmaiQbfaWl
3f6jWfYsea9Oj1ssj9yeYR+A.V6C7ccvd.+PH.h8FJcH.dsGzCib7cUmjF6G
.+aY5QpC4HUmj46azX5RT8qeOSlDRp5vNJ8kMlxQegFIAubSzNqMk67EnP2M
whWhEhj0D19K1IMGvZu1+f8qiX358.hXLmWUmpXFCz40rqHnRYeWA9On9tdx
s2xLqtUfpTr0uLEUBuSJOQXmpSLPkIffVUl3raSiFe3wfOmICctiBc9cmUu5
ncGn.3.D.bBHf9b2.x.7imDbmwsAmwo34MYZdNiS0a5T0cbFEBQSGBQiBgvo
Cgv67dHJ7ZP3rX5LRR.LilNZR.NVRXJPn+UDjXdj5ihBPv4Tp6OFoNZ5rtfA
22PyvQ4QthplBDBFEBwSGBGC9BmQcnQQ.iMveY1l8tRaZb2+pr0+Zrclqv1E
t9Zmb00J63k4x5zOS2lK9UdDS7K5ad0lehxyKSnu5Vh0SVs8vdVbbc8K5ztr
pTyKKuQyUlqc0NPgtPXPQ0NbfN9HO8mTe.iaa1VtHX0pbwtg.Twbc8P9tX8m
BPNXGccS5uNTC1.vPCN.gA.WymTCA6UpEa0YIJuvY9supTIohDQZ8UvasSXy
JxkhCojHVYuvAmzUumpz6RUS4jqLk8Qk3kUuxt5p0aOlSp8iw4zOBu7VCY2N
EZ5Hrcf9dt9ELsWs.LrXnJ1eFnzTwq7afTaHPrRYvMTuSMuj5mRozagT00pC
qKZbwmfdAJcxYiJ+YZzMPi9At5JZqHOWrR0WalnrWPgyFg9qz3Xwq2.sFfvP
sRpe.L.poTbHtvHusIpc6a65m022UbH.DLwrhtvneRDeSBck5JP6sAGhfXMO
Aa6sYJozDBmF+uXhlaIzb4x1o1fTwJ.+.iyWkNu1ML5lbZC+aiS6x36Cbuve
+6TS4+1tUawF
-----------end_max5_patcher-----------
</code></pre>
```

Upon load - no refer, no complaints:
![image](https://user-images.githubusercontent.com/45127463/159161134-5725bf13-cdb3-46df-a863-27c48db0bba7.png)

Upon hitting the button - instantaneous as normally, no problems:
![image](https://user-images.githubusercontent.com/45127463/159161194-df73669f-a258-401d-9951-e466568a0550.png)

### Solution 2: Initialize buffers with @samps 1

Patch:
```
<pre><code>
----------begin_max5_patcher----------
2140.3oc6ajriiai8raf9ePPHGq1fKhZIm5YtjSykY.BvfFCLnsncwzxhBRT
UUcBl9aenH0tsbI6HYmjQUi1xfOR8V3aiuG8u8wOrxdq3MVls0OZ8EqUq9M0
HqziULxppAVYej91tHZldh1wrWEa+E6mJgIYuI0iuMe+dV52sDwYL4t7zWXV
eNidLIyBVO4Dpb2y73CaRY6jFz534rF7jEV+IzMn3ABsFX8epVTb9QdbDSpQ
Or0nhbY0vnpgMiI+VBy75s2GInR6mTzGM9fcyakGpoZEq7Irusdv+6G+PwS0
im9cKMRr3wb4EYbeMKiHAM7+EXbv0x3E7744WTPW5hkVxgUr3J687H1KrzLt
HtsTekMMIo03qZunBIyuHzuK+mZFiGaFC0LVJ6Ed0qn0vzTkLRpDP4oZlv9M
WG6VuIQHKMNmqInxQK1qpHM8lRL8HKKgty7BJ17pg2H5MhbHvs3gqCn3giOn
krWsueHRr6qrv1xdkTNgEyiSRYYrXIUVxCMvCY6o4QxM6EwxL9upoBnZS8rS
XeIYddnEbhlI9aobZTCabHkGJhKHjt6LEiWgxuXAIZVjzgmzSIllbtkqTLTx
mgfloX17rszzhMtsQrN6b1RgHpKrlUFw1KKgmvii6KPkhjK.Mke34Ks5sBEz
iW70qAksIO1.diRIQtIi9ROIujFEUZK2CCuQi4GoRlja1OPfFnrXphgeNaWp
HJpKea.8x4.EpT+2wdkGJeVirNpGpEvSpTrra11C4GXYxdCJoGx5MTl7alsf
1ikuszPeijcLIRwM8lgxxgmIydV7ZV4LqT+5HJZhUzwvusWxt.tn2xtdLU9o
C09naC8L9MQ9FUaf1rk3eheygCZbF+mc.chOztwLNwOpicMfxXGU9jtYIjx6
UF8.67hnTlJBqUXZ9wmrTtJ3wrTEAxr9DrIF66I1JeLJ4F5VkaWPlguyxLib
JQvikVwhXVkjS9Le2WiUqTwJkCsSDIRs.5+UOXj5+V5zYdOILQqPBMgSvA3G
j.FL8BXC5GG+ib8z7On5yaytDLD6Q5PF6DGOphD20aldIwgr2Zmov8VZTZuM
2hC2qTbfFPbzJKgBM9g7yqYzhILffJSjmtqRIszG4SV8XZUPNIOtN6ouT6Y3
jYJRCMImBF+t3bRhfKPhv4jDwilDcMybNIFvnIFxEIlJ.MIwaWjXV3FStnan
RYJeatznM147IWYdRpb42Ri5kYyYyiplht4SDlDIje+RG7ifLm30GpO2AT6f
vkbsG4EN7I+N+obA0DUQ9fJprsQdyg3zQAMPzNsZ9.1w6kccDTM0zjDqhoKh
1toISzh3tsNHWgxVKn8BtVvlaUYpN3xUmMUZL45CITbjxiUqjlJ6kYcMPVb3
Iza85pvXAERSO8Mejl9U0oeMRm1xhR3Qzsrny.VmlVCc8oNX2.7bzU0xFjrL
SXPpx.dHhRnNckhle6DIUIfuUBn4n1kFrSZcRRYpCaF1Nc1KY63FfVSpKZBD
dikK5pscf2dEh5kcZCq+CDqevaLLqgKwuGuhlHdEM8UCaeTNOb8178nH0wLG
AOCMNG8PSMOqw+44a7Lx2E0OI66Ve1DLsUMRGirvwTaz.2aSXbKkHD6L8Bih
C3WXLXYNF63jAkAKqL3KOiw7avSrm+D.fN22D.bWR.3OJI.3u1wASBb8wAA.
GOex.oCPvqQHGB10Of3QfAD7+WkcPs+yFmEMNQKpCl0m2ynE8K35bmDTZ.hu
MuI3As7LEvrUWmFvTz6QzwoR18AzvIukFNszvokFNszvokFNszvokFNszvok
FNszvokFNszvokFNcOZ3D9QzuIDbobSK8a5O48aBgfOh9MA8eD8aphYuu8aB
Rdj8apjme.8aBBte8ahdL4cKMZkjnpaSf6W2lBldIQ0uFgJNeT+VDPJs.21+
4617KS.+.9kI3O6cgaLZFj10MtT+X98ANqcfC+PZ.2RBQ+gIgH.XsGvyASbA
dAP.j3NTBR.xZWnKAg8bTmsl34C+KYBSpi8HUms48MZLl+0e9dlLIzT0weT5
KaLEn9BsVBd41pcVaJm4Kzgt+hEeDIDIqo78Wr2Z.2031+gBpCc339.BcLm2
hmpfFCzK1rqHpRYmXAdOnNwdxE6xLqt0jpTr0uvEUBuSJXQPm5ULPsJffV0p
3raSiFejwfO7jgNmQgNutypWk0tCT.b.B.NADPeta.Y.4wSBNy31.dbJdtSl
lGdbpdSmpNFOJDhlNDhFEBgSGBg248PTv0fvYwzYjj.XFMczj.brjvTfPuqH
Hw7H0GEEffyoT2aLRczzYcA8uuglgixibEUMEHDLJDRlNDNF7ELi5Pih.Faf
+xrM6cI2z3t+kaq+Ea6LWpsKbg1N4xrU1CLy02oeltMWEr7Pt3eouKVa9Gr3
7xD5qt2X8jUaOrmGEUW+hNMPqJ07xxazbI5ZWsCTfCD5WTsCLD6gb0eS8EBo
sYa4hfUqxg3D.PEy0wE44PzeyGgIXccS5uNTC1.v.CN.A9.Gy2TCA6UpEa0Y
IJuBZdsu7TIohDQZ8kxaMNnYE4RwgTZHur63fS5y2SU5copobxknx9nR7xqW
YWc05sGyI096Q4rOAu7VCc2NEZ5HrwPOWGuBl1sV.FTLTE6OCTZp303afTaH
PhRYvIPuSMuj5OkxX2BopqUGQWBfhuAc8U5jyFU9OYg2.M546fgZ0cnCQo5q
MST1KE0rXlHz+MKJR75MPq9HBTqj54C8gZJkDPJLxaahZ299u9E8MfkD..9S
LqnKL5OIhtIgtRcEn81PBPPhlmfs81LkTZBMlE8ybQy8FZtbYiqMHUrBvy23
7UoyqcCitIm1v+x3ztL99.2T7O9A0T9e.CU7YcA
-----------end_max5_patcher-----------
</code></pre>
```

Upon load - not empty, no complaints:
![image](https://user-images.githubusercontent.com/45127463/159161299-38cf35ba-a939-4cb3-b7f7-28dec8c18ed1.png)

Upon hitting the button - instantaneous as normally, no problems:
![image](https://user-images.githubusercontent.com/45127463/159161319-3d6da959-80fa-496c-afb7-97776bd89f3c.png)

**Comment by weefuzzy** [comment]
I'm not sure quite sure what causes this, beyond being pretty sure it's `plot~` and not us. Seems like if `plot~` is induced to produce the error message `plot~: buffer~ <X> has no samples` then it doesn't release its lock on the buffer, and we're stuck. However, the conditions for getting `plot~` to say this are unclear: if I resize a buffer to `0 0`, then it doesn't complain, but it does with a completely fresh buffer. 

So, tl;dr: not our bug, as far as I'm concerned 😄

**Comment by balintlaczko** [comment]
Fair enough! I see that when the buffer is new, the `plot~` complains only the first time, and never at subsequent `refer` messages, although the buffer remains the same "fresh" buffer. This indicates to me that it subscribes to the buffer, since immediately after I call `samps 10, fill cos 2` on the buffer the `plot~` will redraw itself, probably due to some `notify_changed` event firing under the hood.
I guess when I am saying `samps N` to the buffer it does not extend it rather replace it with a new empty buffer of that length (with the same identifier name).
On the other hand if I substitute the `plot~` with a `waveform~` the problem does not happen, so it's clearly something _funky_ about `plot~` (like the interface was not funky enough already :).
